### DISCUSSION TITLE
FYI: cp2k.org/gpu

### DISCUSSION THREAD

**Discussion by oschuett** [cp2k/cp2k#2608] [Team Posts]
I've started a wiki page to track the status of our various GPU efforts: https://www.cp2k.org/gpu

**Comment by alazzaro** [comment]
Better add a column to say which technology we support (CUDA, HIP...)

**Comment by robertschade** [comment]
Could we have a column that lists the people/groups that can be contacted?

**Comment by oschuett** [comment]
> Better add a column to say which technology we support (CUDA, HIP...)

Good idea! While most of the libraries support more backends, I only listed those that we have already validated and are actively using via the toolchain.

> Could we have a column that lists the people/groups that can be contacted?

For the existing projects one can just follow the links and the "planned" projects are really up for grabs.

The new eigensolver will probably become part of SPLA, right @mtaillefumier?

As for the ERIs, according to my latest information this will be a collaboration between PC2 and Lincoln. Maybe there is already a project page I could link to?

By the way, feel free to add more "planned" projects if you think that an important workload is missing.

**Comment by mtaillefumier** [comment]
I also think it is a good idea to regroup information about what is GPU supported. I will reply to filippo on Monday. 
The new eigensolver is still I would say not ready for production. We will most probably not include it right away to spla (I do not even recall it would be part of it) but it might be worth discussing when the eigensolver is ready for use.
I will update the sirius side of things.

**Comment by robertschade** [comment]
@oschuett Who is working on two-electron integrals on GPUs?

**Comment by oschuett** [comment]
> Who is working on two-electron integrals on GPUs?

Well, I thought it's PC2 (ie. you ;-) and @mattatlincoln.

**Comment by robertschade** [comment]
Ah, ok. We are currently focussing on the FPGA implementation but are considering using the things that we have learned especially in our exploration of the available algorithms also to GPUs.

**Comment by robertschade** [comment]
Thus, from our side, it would be better to call it "Planned".

**Comment by mattatlincoln** [comment]
Hello,
yes, we have just started work on 2c integrals. Would be good to check on best ways forward with others working on related areas.
There is also other generic GPU started in UK that we can feed into if there are useful ideas.
Matt

**Comment by mattatlincoln** [comment]
Crossed emails. Ours is also in the 'planned' or in planning stage algorithmically.

**Comment by Growl1234** [comment]
Should we migrate it to GitHub wiki?

**Reply by he-zilong** [reply]
Either github wiki or manual documentation for the supported GPU features may be fine. As for WIP or planned ones, they may be merged with [dev:todo](https://www.cp2k.org/dev:todo) (which can be migrated to github wiki/discussions/issues).

**Reply by Growl1234** [reply]
That page has not been updated for several years. I am not sure how we can take better use of it, probably together with things like GPU support status and new feature requests.

**Comment by oschuett** [comment]
Yes, let's continue to clear out the old wiki.

The [GPU page]( https://www.cp2k.org/gpu) could be moved to the [accelerators section](https://manual.cp2k.org/trunk/technologies/accelerators/index.html) of the manual.

Whatever is still relevant of [dev:todo](https://www.cp2k.org/dev:todo) could be turned into Github issues.
